### PR TITLE
Add more JSDoc types

### PR DESCRIPTION
**Pull request by kungfooman**
I like to typify code 😅 I don't know how self-explanatory these types are, I happily answer questions.

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

**Comment by willeastcott** [comment]
Just a nudge to say that the CI is failing...

**Comment by willeastcott** [comment]
Can you confirm that `npm run docs` completes with no errors? I need to add that to the GitHub Actions....

**Comment by kungfooman** [comment]
> Can you confirm that `npm run docs` completes with no errors? I need to add that to the GitHub Actions....

There are eight errors, two related to this PR (should have used `@callback`)

Currently rewriting this PR to use `@callback` style instead of TS inline callback type

File: src/framework/app-base.js [hunk 1]
```diff
@@ -2078,17 +2078,31 @@ class AppBase extends EventHandler {
...
 const _frameEndData = {};
 
+/**
+ * Callback used by {@link AppBase#start} and itself to request
+ * the rendering of a new animation frame.
+ * 
+ * @callback MakeTickCallback
+ * @param {number} [timestamp] - The timestamp supplied by requestAnimationFrame.
+ * @param {*} [frame] - XRFrame from requestAnimationFrame callback.
+ * @returns {void}
```
Comment (willeastcott) [inline src/framework/app-base.js:2088]: I think you can delete this line right?

Comment (kungfooman) [reply src/framework/app-base.js]: Yep, thank you!

**Comment by kungfooman** [comment]
I tried to look into the other errors and it boils down to `typeof` not being supported:

https://github.com/playcanvas/engine/blob/e6d399454c42c15370de205d187a597ea93bd8a6/src/framework/app-options.js#L85-L104

However, I don't know a way to fix it in pure JSDoc. It seems like e.g. ` @lends {XrManager}` is just the right solution, but VSCode/TypeScript don't support that:

https://github.com/microsoft/TypeScript/issues/36771

What VSCode/TypeScript likes most seems to be: `@type {{new(app: AppBase): XrManager}}`

(basically for the lack of a proper `ConstructorOf<XrManager>` generic type)

But that trips up JSDoc again... so I feel like `@lends` is still the most correct way (but TypeScript doesn't supports it yet)

Doc on `@lends`: https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#lends-objectname

**Review by willeastcott — APPROVED** [review]

**Pull request merged**